trio/_tests/test_exports.py

```diff
@@ -118,18 +116,45 @@ def no_underscores(symbols):
         )
         from mypy.api import run
 
-        res = run(["--config-file=", "--follow-imports=silent", str(tmpfile)])
-
-        # clean up created py.typed file
-        if not py_typed_exists:  # pragma: no cover
-            py_typed_path.unlink()
+        mypy_res = run(["--config-file=", "--follow-imports=silent", str(tmpfile)])
 
         # check that there were no errors (exit code 0), otherwise print the errors
-        assert res[2] == 0, res[0]
+        assert mypy_res[2] == 0, mypy_res[0]
         return
+    elif tool == "pyright_verifytypes":
+        import subprocess
+
+        res = subprocess.run(
+            ["pyright", f"--verifytypes={modname}", "--verbose"],
+            capture_output=True,
+        )
+        start_index = res.stdout.find(b"Public modules: ")
+        end_index = res.stdout.find(b"Other referenced symbols: ", start_index)
+        assert start_index != -1, (res.stdout, res.stderr)
+        assert end_index != -1, res.stdout
+        static_names = {
+            x.strip().decode()[len(modname) + 1 :]
+            for x in res.stdout[start_index:end_index].split(b"\n")
+            if modname + "." in x.decode()
+        }
+
+        # don't require verifytypes to see reexported constants from socket
```

Comment:
I disagree with this comment but if it's annoying we can bump this off

Reply:
wait no. hmm

Reply:
> Type annotations can be omitted in a few specific cases where the type is obvious from the context:
> * ...
> * Constants that are assigned simple literal values (e.g. RED = '#F00' or MAX_TIMEOUT = 50 or room_temperature: Final = 20). A constant is a symbol that is assigned only once and is either annotated with Final or is named in all-caps. A constant that is not assigned a simple literal value requires explicit annotations, preferably with a Final annotation (e.g. WOODWINDS: Final[list[str]] = ['Oboe', 'Bassoon']).
https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#type-completeness

So it's plausibly a bug that they don't show up in `--verbose` or `--outputjson`, but it's intended that they're not covered by `--verifytypes`.

Reply:
Actually no, it's the usual
> Imported symbols are considered private by default. If they use the “import A as A” (a redundant module alias), “from X import A as A” (a redundant symbol alias), or “from . import A” forms, symbol “A” is not private unless the name begins with an underscore. 

that hits here. So what'd need to be changed here is changing the huge list of imports from `socket` to be `from socket import (X as X, Y as Y, Z as Z)`. 

I don't think we care for this in terms of `--verifytypes`, *but* if somebody uses pyright as their type checker it will give an error of `reportPrivateImportUsage` when they try accessing it. That seems important enough to warrant making that huge bulk of an import statement even messier

Reply:
Hm yeah, well it's a gigantic change so let's maybe sit on it a bit! Maybe an ad-hoc codemod would help out a ton, in combination with a run from `black`?

Reply:
vim macros FTW :D